crates/common/src/signer/schemes/bls.rs

```diff
@@ -0,0 +1,100 @@
+use alloy::rpc::types::beacon::{
+    constants::BLS_DST_SIG, BlsPublicKey as BlsPublicKeyInner, BlsSignature,
+};
+use blst::BLST_ERROR;
+use derive_more::derive::{Deref, Display, From, Into, LowerHex};
+use serde::{Deserialize, Serialize};
+use ssz_derive::{Decode, Encode};
+use tree_hash::TreeHash;
+use tree_hash_derive::TreeHash;
+
+use crate::{
+    error::BlstErrorWrapper, signature::sign_builder_root, types::Chain,
+    utils::blst_pubkey_to_alloy,
+};
+
+pub type BlsSecretKey = blst::min_pk::SecretKey;
+
+// std traits
+#[derive(Debug, Clone, Copy, LowerHex, Display, PartialEq, Eq, Hash, Default)]
+// serde, ssz, tree_hash
+#[derive(Serialize, Deserialize, Encode, Decode, TreeHash)]
+#[ssz(struct_behaviour = "transparent")]
+#[serde(transparent)]
+// derive_more
+#[derive(Deref, From, Into)]
+pub struct BlsPublicKey {
+    inner: BlsPublicKeyInner,
+}
+
+impl AsRef<[u8]> for BlsPublicKey {
+    fn as_ref(&self) -> &[u8] {
+        self.as_slice()
+    }
+}
+
+#[derive(Clone)]
+pub enum BlsSigner {
```

Comment:
not entirely sure, why an enum with only one variant?

Reply:
That's inherited from what was previously set up in the old bare `Signer`; the other variant would be a remote signer (to be implemented in future). That's also the reason why some of the methods are async (although they're synchronous for the local variant).

Reply:
ok, fair enough. In general I prefer to simplify code where possible but remote seems something we will implement sooner rather than later so let's keep as is